### PR TITLE
Fixed character types for Windows to be universal

### PR DESCRIPTION
**Pull request by maximmasiutin**
This pull requests addresses the issues described in resolved in https://github.com/official-stockfish/Stockfish/pull/4438 with earlier code, but new code appeared that should also be fixed.

### CODE REVIEW

**Comment by noobpwnftw** [comment]
`fun7_t` should remain ansi since explicitly requested it.

**Comment by vondele** [comment]
so, @noobpwnftw I understand correctly this PR is not needed, and with https://github.com/official-stockfish/Stockfish/pull/4507 we're correct? 

(not having windows, nor particularly interested in it, I need to rely on others to review this kind of PRs)

**Comment by noobpwnftw** [comment]
right

**Comment by dubslow** [comment]
> (not having windows, nor particularly interested in it)

(a happy boat to be in)

**Comment by maximmasiutin** [comment]
Both pull requests are needed. One resolves ASCII issues, another is refactoring (changes variable name to one explicit without affecting binary code)

File: src/misc.cpp [hunk 1]
```diff
@@ -39,7 +39,7 @@ using fun3_t = bool(*)(HANDLE, CONST GROUP_AFFINITY*, PGROUP_AFFINITY);
 using fun4_t = bool(*)(USHORT, PGROUP_AFFINITY, USHORT, PUSHORT);
 using fun5_t = WORD(*)();
 using fun6_t = bool(*)(HANDLE, DWORD, PHANDLE);
-using fun7_t = bool(*)(LPCSTR, LPCSTR, PLUID);
+using fun7_t = bool(*)(LPCTSTR, LPCTSTR, PLUID);
```
Comment (noobpwnftw) [inline src/misc.cpp:42]: This is incorrect. We explicitly asked for "LookupPrivilegeValueA" which is ANSI.

**Comment by maximmasiutin** [comment]
I made rebase+squash.

Can you please reopen it as I have resolved observations by @noobpwnftw and took code by @noobpwnftw related to the variable name that holds the module handle.

To @noobpwnftw -- if you support these changes, can you please close your pull request https://github.com/official-stockfish/Stockfish/pull/4507 and comment that this one resolves the issue?

**Comment by maximmasiutin** [comment]
This one is closed in favor of https://github.com/official-stockfish/Stockfish/pull/4509

**Pull request closed without merging**